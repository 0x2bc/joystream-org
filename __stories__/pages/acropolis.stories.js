import React from 'react';
import { storiesOf } from '@storybook/react';

import AcropolisPage from '../../src/pages/acropolis';

storiesOf('Pages|Acropolis', module).add('default', () => <AcropolisPage />);
