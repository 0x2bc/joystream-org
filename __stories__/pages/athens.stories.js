import React from 'react';
import { storiesOf } from '@storybook/react';

import AthensPage from '../../src/pages/athens';

storiesOf('Pages|Athens', module).add('default', () => <AthensPage />);
