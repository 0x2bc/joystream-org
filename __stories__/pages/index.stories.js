import React from 'react';
import { storiesOf } from '@storybook/react';

import IndexPage from '../../src/pages/index';

storiesOf('Pages|Home', module).add('default', () => <IndexPage />);
