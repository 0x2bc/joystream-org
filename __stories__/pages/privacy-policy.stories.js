import React from 'react';
import { storiesOf } from '@storybook/react';

import PrivacyPolicyPage from '../../src/pages/privacy-policy';

storiesOf('Pages|Privacy Policy', module).add('default', () => <PrivacyPolicyPage />);
