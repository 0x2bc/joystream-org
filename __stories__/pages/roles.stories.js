import React from 'react';
import { storiesOf } from '@storybook/react';

import RolesPage from '../../src/pages/roles';

storiesOf('Pages|Roles', module).add('default', () => <RolesPage />);
