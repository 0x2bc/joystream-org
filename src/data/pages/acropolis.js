const goalsData = [
  {
    title: 'Build and release an on-chain forum',
    text: `The final platform will have built in multiple ways of facilitating
            easy and secure public, and private, communication and information
            sharing among members. The forum is the first step allowing the
            platform members to communicate, share ideas and discuss.`,
  },
  {
    title: 'Rebuild and release the storage node',
    text: `Where the old storage node had some bugs making it unable to sync
            between clients and required a hardcoded liaison, the new storage
            node is working as intended with no hierarchy or privileges.`,
  },
  {
    title: 'Ensure high uptime of storage providers',
    text: `Both content creators and consumers user experience depends on
            various metrics, but perhaps the most important is that the storage
            node they connect to responds to their request. Whether or not this
            goal is achieved depends on both the reliability of the software, a
            good reporting system and corrective actions from multiple parties.`,
    state: 'inprogress',
  },
  {
    title: 'Build the storage node with "tranches"',
    text: `Although the content directory is currently small, at some point
            it's not feasible to expect all storage providers to have full
            replication of all the content. Tranches would allow storage
            providers to select a subset of the content directory, suitable to
            their capabilities.`,
    state: 'postponed',
  },
];

export { goalsData };
