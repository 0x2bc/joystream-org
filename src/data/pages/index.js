import { ReactComponent as ValidatorsImage } from '../../assets/svg/active-validators.svg';
import { ReactComponent as StorageImage } from '../../assets/svg/storage.svg';
import { ReactComponent as MemberImage } from '../../assets/svg/council-member.svg';
import { ReactComponent as ScreenerImage } from '../../assets/svg/screener.svg';
import { ReactComponent as MembershipCuratorImage } from '../../assets/svg/membership-curator.svg';
import { ReactComponent as ContentCuratorImage } from '../../assets/svg/content-curator.svg';
import { ReactComponent as ContentCreatorImage } from '../../assets/svg/content-creator.svg';
import { ReactComponent as BandwidthProviderImage } from '../../assets/svg/bandwidth-provider.svg';
import { ReactComponent as DiscoveryProviderImage } from '../../assets/svg/discovery-provider.svg';
import { ReactComponent as LiveStreamingProviderImage } from '../../assets/svg/live-streaming-provider.svg';
import { ReactComponent as BuilderImage } from '../../assets/svg/builder.svg';
import { ReactComponent as CommunicationModeratorImage } from '../../assets/svg/communication-moderator.svg';

const roles = {
  active: [
    {
      image: ValidatorsImage,
      title: 'Validator',
      to: '/roles#Validator',
      key: 'validatorsCount',
      hasLabel: false,
    },
    {
      image: MemberImage,
      title: 'Council Member',
      to: '/roles#Council-Member',
      key: 'councilMembersCount',
      hasLabel: false,
    },
    {
      image: StorageImage,
      title: 'Storage Provider',
      to: '/roles#Storage-Provider',
      key: 'storageProviders',
      hasLabel: true,
    },
  ],
  future: [
    { image: ScreenerImage, title: 'Membership Screener', to: '/roles' },
    {
      image: MembershipCuratorImage,
      title: 'Membership Curator',
      to: '/roles',
    },
    { image: ContentCuratorImage, title: 'Content Curator', to: '/roles' },
    { image: ContentCreatorImage, title: 'Content Creator', to: '/roles' },
    {
      image: BandwidthProviderImage,
      title: 'Bandwidth Provider',
      to: '/roles',
    },
    {
      image: DiscoveryProviderImage,
      title: 'Discovery Provider',
      to: '/roles',
    },
    {
      image: LiveStreamingProviderImage,
      title: 'Live Streaming Provider',
      to: '/roles',
    },
    { image: BuilderImage, title: 'Builder', to: '/roles' },
    {
      image: CommunicationModeratorImage,
      title: 'Communication Moderator',
      to: '/roles',
    },
  ],
};

export { roles };
