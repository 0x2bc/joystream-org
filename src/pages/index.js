import React from 'react';
import { pagePropTypes } from '../propTypes';

import getApiPath from '../utils/getApiPath';
import mapStatusDataToAnalytics from '../utils/mapStatusDataToAnalytics';
import mapStatusDataToRoles from '../utils/mapStatusDataToRoles';

import withApi from '../components/_enhancers/withApi';

import BaseLayout from '../components/_layouts/Base';
import Link from '../components/Link';
import Subheader from '../components/Subheader';
import TestnetItem from '../components/TestnetItem';
import Analytics from '../components/Analytics';
import Button from '../components/Button';
import TitleWrapper from '../components/TitleWrapper';
import RoleList from '../components/RoleList';
import ColumnsLayout from '../components/ColumnsLayout';
import LayoutWrapper from '../components/LayoutWrapper';
import Hero from '../components/Hero';
import SiteMetadata from '../components/SiteMetadata';

import AcropolisImage from '../assets/svg/acropolis-main.svg';
import heroImage from '../assets/svg/hero-builder.svg';
import RomeImage from '../assets/svg/rome-main.svg';
import constantinopleImage from '../assets/svg/constantinople-main.svg';
import alexandriaImage from '../assets/svg/alexandria-main.svg';
import { ReactComponent as TickImage } from '../assets/svg/tick.svg';
import { ReactComponent as ClockImage } from '../assets/svg/clock.svg';

import { roles } from '../data/pages';
import { launchDate as alexandriaNetworkLaunchDate } from '../data/pages/alexandria';

import './style.scss';

const activeTestnet = {
  name: 'Alexandria',
  incentivesLink: 'https://www.joystream.org/testnet',
};

const IndexPage = ({ content }) => (
  <BaseLayout>
    <SiteMetadata
      title="Joystream: The video platform DAO"
      description="Joystream is a video platform controlled, owned, and operated by its users."
    />
    <Hero
      image={heroImage}
      title={
        <>
          The video <br /> platform DAO
        </>
      }
      animationStartValue={0}
    >
      <p className="IndexPage__hero-paragraph">
        Joystream is a video platform controlled, owned, and operated by its users.
      </p>
      <div className="IndexPage__hero-group">
        <Button noWrap large secondary className="IndexPage__hero-button" href="/testnet/">
          Earn Tokens
        </Button>
        <Button
          noWrap
          large
          secondary
          reversed
          className="IndexPage__hero-button"
          href="https://testnet.joystream.org/"
        >
          Launch UI
        </Button>
      </div>
    </Hero>

    <LayoutWrapper>
      <TestnetItem
        title="Alexandria Testnet"
        image={alexandriaImage}
        children={
          <>Alexandria is our seventh testnet, introducing a variety of important technical improvements.</>
        }
        button={{
          label: 'Explore Alexandria',
          to: '/alexandria',
        }}
      />
    </LayoutWrapper>

    <Analytics title="Testnet Metrics" large content={mapStatusDataToAnalytics(content)}>
      <Button secondary href={activeTestnet.incentivesLink}>
        Participate and Earn Monero
      </Button>
    </Analytics>

    {/*

    <LayoutWrapper>
      <TestnetItem
        title="Alexandria Testnet"
        image={alexandriaImage}
        children="Alexandria is our seventh testnet, focused on a variety of important technical improvements."
        date={alexandriaNetworkLaunchDate}
        button={{
          label: 'Explore Alexandria',
          to: '/alexandria',
        }}
      />
    </LayoutWrapper>

    */}

    <LayoutWrapper dark>
      <TitleWrapper title="Become a user, run the platform">
        <Subheader
          title="Active roles on the current testnet"
          content="learn more, join a role and subscribe for more"
          icon={TickImage}
        />
        <ColumnsLayout>
          <RoleList roles={roles.active} content={mapStatusDataToRoles(content)} />
        </ColumnsLayout>
        <Subheader
          title="Roles coming in future testnets"
          content="choose a role, learn more and subscribe to get updated"
          icon={ClockImage}
        />
        <ColumnsLayout columnsCount={3}>
          <RoleList roles={roles.future} content={mapStatusDataToRoles(content)} />
        </ColumnsLayout>

        <Button secondary to="/roles" className="IndexPage__roles-button">
          Learn more about the roles
        </Button>
      </TitleWrapper>
    </LayoutWrapper>
  </BaseLayout>
);

IndexPage.propTypes = pagePropTypes;

export { IndexPage };
export default withApi(IndexPage, getApiPath('/'));
