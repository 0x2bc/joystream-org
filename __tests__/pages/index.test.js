import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { IndexPage } from '../../src/pages';

import testContent from '../../__mocks__/data/testContent';

configure({ adapter: new Adapter() });

describe('IndexPage page', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<IndexPage content={testContent} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
