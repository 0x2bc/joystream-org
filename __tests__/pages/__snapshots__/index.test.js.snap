// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage page renders correctly 1`] = `
<BaseLayout
  t={[Function]}
>
  <SiteMetadata
    description="landing.siteMetadata.description"
    image="test-file-stub"
    lang="en"
    meta={Array []}
    title="siteMetadata.title"
  />
  <Hero
    statusData={null}
    t={[Function]}
  />
  <EarnJOYCoins
    t={[Function]}
  />
  <BecomeFoundingMember
    t={[Function]}
  />
  <WhatWeDo
    t={[Function]}
  />
  <WhyYouShouldJoin
    t={[Function]}
  />
  <ExploreJoystream
    t={[Function]}
  />
  <RoadToMainnet
    t={[Function]}
  />
  <EarnTokens
    t={[Function]}
  />
  <OurInvestors
    t={[Function]}
  />
</BaseLayout>
`;
